fix: LEAP-1123: Fix the skip button when the annotator doesn't leave a comment

juliosgarbi opened the pull request:
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [x] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)



#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend



### Describe the reason for change
This PR updates the comment validation logic to properly handle the transition of currentComment from a string to an object. Previously, the validation was based on currentComment.trim(), assuming it was a string. Now, the validation will check for objects within currentComment.
The currentComment field has changed from a string to an object. The existing validation logic did not account for this change, leading to incorrect validation behavior.



#### What does this fix?
Stops validating currentComment as a trimmed string.
Starts validating currentComment based on whether it contains objects.
Ensures that when a user tries to skip a annotation, the currentComment correctly validates new objects inside it.


### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure (briefly explain the situation below)



### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [x] unit

juliosgarbi committed:
fix: LEAP-1123: Fix skip button when annotator didn't leave a comment

juliosgarbi committed:
fix typo and remove unused var

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9472713996

hlomzik commented on web/libs/editor/src/components/BottomBar/Controls.jsx:
```diff
@@ -60,7 +59,7 @@ export const Controls = controlsInjector(
           selected?.submissionInProgress();
           callback();
-        } else if ((currentComment ?? "").trim()) {
+        } else if (Object.keys(currentComment).length > 0) {
```
I think, we should check only current comment for current annotation. The rest could still stay there

juliosgarbi replied:
Thanks @hlomzik, I miss understood the way that we were validating this! It's fixed!

juliosgarbi committed:
fix skip just when its the right ann id

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9473262997

juliosgarbi committed:
add tests for empty message

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9473310153

juliosgarbi committed:
add more comments to check if it's working even with comments from another annotations

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9473376211

juliosgarbi committed:
fix biome

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9473426275

juliosgarbi commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/9486723278)
> Successfully merged:  create mode 100644 label_studio/io_storages/s3/openapi_schema.py

juliosgarbi committed:
Merge branch 'develop' into 'fb-leap-1123'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9486723278

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/9486748992
